File: tests/wasm2lang_04_arithmetic.harness.java
// Java test harness for wasm2lang_04_arithmetic.
{
    java.util.Map<String, Object> foreign = new java.util.LinkedHashMap<>();

    WasmModule mod = new WasmModule(foreign, memBuffer);
    mod.alignHeapTop();

    java.util.Map<String, Object> _data = w2lLoadSharedData(System.getProperty("w2l.testname", ""));

    for (Double v : w2lFlat(_data, "i32_values")) {
        mod.exerciseNestedArithmetic(v.intValue());
    }

    for (java.util.List<Double> p : w2lNested(_data, "i32_pairs")) {
        mod.exerciseMemoryArithmetic(p.get(0).intValue(), p.get(1).intValue());
    }

    java.util.List<java.util.List<Double>> _mtc = w2lNested(_data, "mixed_type_cases");
    for (int _i = 0; _i < 4 && _i < _mtc.size(); ++_i) {
        java.util.List<Double> t = _mtc.get(_i);
        mod.exerciseMixedTypeChains(t.get(0).intValue(), t.get(1).floatValue(), t.get(2));
    }

    mod.exerciseEdgeArithmetic();

    w2lDumpCRC(memBuffer);
}

/exit

File: tests/wasm2lang_04_arithmetic.harness.php
<?php
declare(strict_types=1);

$moduleImports = [];

$runTest = function (string &$buff, callable $out, array $exports, ?array $data = null) use (
    &$stdoutWrite
): void {
    $stdoutWrite = $out;
    $exports['alignHeapTop']();

    foreach ($data['i32_values'] as $a) {
        $exports['exerciseNestedArithmetic']($a);
    }

    foreach ($data['i32_pairs'] as $scenario) {
        $exports['exerciseMemoryArithmetic']($scenario[0], $scenario[1]);
    }

    foreach (array_slice($data['mixed_type_cases'], 0, 4) as $scenario) {
        $exports['exerciseMixedTypeChains']($scenario[0], $scenario[1], $scenario[2]);
    }

    $exports['exerciseEdgeArithmetic']();
};

$dumpMemory = true;
